android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java

```diff
@@ -31,7 +38,40 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
       return;
     }
 
+    ExponentDB.experienceIdToExperience(remoteMessage.getData().get("experienceId"), new ExponentDB.ExperienceResultListener() {
+      @Override
+      public void onSuccess(ExperienceDBObject experience) {
+        try {
+          JSONObject manifest = new JSONObject(experience.manifest);
+          JSONObject androidSection = manifest.optJSONObject("android");
+          if (androidSection != null) {
+            boolean useNewNotificationsAPI = androidSection.optBoolean("useNewNotificationsAPI", false);
+            if (useNewNotificationsAPI) {
+              dispatchToNewNotificationModule(remoteMessage);
+              return;
+            }
+          }
+          dispatchToOldNotificationModule(remoteMessage);
+        } catch (JSONException e) {
+          e.printStackTrace();
+          onFailure();
+        }
+      }
+
+      @Override
+      public void onFailure() {
+        Log.w("expo-notifications", "Couldn't get experience from remote message.", null);
+        dispatchToNewNotificationModule(remoteMessage);
+        dispatchToOldNotificationModule(remoteMessage);
+      }
```

Comment:
I'm not sure what should we do here 🤔

Reply:
Me too 🤔 Let's dispatch only to the old notifications module. 🤷 If no `useNewNotificationsAPI` falls back to using the old module, no manifest should fall back the same.